src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -15,36 +15,53 @@ export class TextInputComponent implements OnInit {
 
   showDatePicker: boolean
   showTimePicker: boolean
+  showDurationPicker: boolean
+  showTextInput
+
   showSeconds: boolean
   dateFormat: string
+  datePickerValues: string[]
   timeFormat: string
-  customMonthNames = []
-  datetimeValue: { [key: string]: any } = {}
+  timePickerValues: string[]
+  durationPickerValues: string[]
+
+  value = {}
 
   constructor(private localization: LocalizationService) {}
 
   ngOnInit() {
     this.showDatePicker = this.type.includes('date')
     this.showTimePicker = this.type.includes('time')
+    this.showDurationPicker = this.type.includes('duration')
+    this.showTextInput = this.type.includes('text')
     this.showSeconds = this.type.includes('seconds')
-    this.initFormats()
+    this.initValues()
   }
 
-  initFormats() {
+  initValues() {
     const locale = this.localization.moment().localeData()
-    this.dateFormat = locale.longDateFormat('LL').replace('MMMM', 'MMM')
-    this.timeFormat = this.showSeconds
-      ? locale.longDateFormat('LTS')
-      : locale.longDateFormat('LT')
-    this.customMonthNames = locale.monthsShort()
+    const months = locale.monthsShort()
+    const days = this.addLeadingZero(Array.from(Array(32).keys()).slice(1, 32))
+    const years = Array.from(Array(21).keys()).map(d => d + 2000)
+    this.datePickerValues = [months, days, years]
+
+    const hours = this.addLeadingZero(Array.from(Array(13).keys()).slice(1, 13))
+    const minutes = this.addLeadingZero(
+      Array.from(Array(60).keys()).slice(1, 60)
+    )
+    const meridiem = ['AM', 'PM']
+    this.timePickerValues = [hours, minutes, meridiem]
+    if (this.showSeconds) this.timePickerValues.push(minutes)
+
+    const duration = this.addLeadingZero(Array.from(Array(100).keys()))
+    this.durationPickerValues = [duration, duration]
```

Comment:
can probably just use `[hours, minutes]`, since if treated as duration, a selection of e.g. `89:75` wouldn't make sense here.
And then maybe go up to 23h? Not necessary for our purpose but might be nice for other possible future uses.